src/main/java/com/uber/cadence/internal/worker/ActivityWorker.java

```diff
@@ -157,26 +160,31 @@ private TaskHandlerImpl(ActivityTaskHandler handler) {
 
     @Override
     public void handle(MeasurableActivityTask task) throws Exception {
-      options
-          .getMetricsScope()
+      Scope metricsScope =
+          options
+              .getMetricsScope()
+              .tagged(
+                  ImmutableMap.of(MetricsTag.ACTIVITY_TYPE, task.task.getActivityType().getName()));
```

Comment:
Can we also add the WORKFLOW_TYPE metrics tag?

Reply:
Activity worker does not have that info. Workflow worker has that in decision task completion.